source/common/upstream/health_checker_impl.cc

```diff
@@ -418,6 +420,7 @@ void TcpHealthCheckerImpl::TcpActiveHealthCheckSession::onInterval() {
 
     client_->connect();
     client_->noDelay(true);
+    expect_reset_ = false;
```

Comment:
I think I would move this before the connect() call in case there is an inline reset? I'm not sure if that can happen or not but might as well?

Reply:
Sounds good, fixed.